### PR TITLE
Check for a two part, protocol provided, addr

### LINKED ISSUE #72: Validate client address arg
client, err := abciclient.NewClient("127.0.0.1:46657", "socket", true)

```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/tendermint/tendermint/vendor/github.com/tendermint/go-common.Connect(0x87bdd3, 0xf, 0x4470ed, 0xc4200b7be8, 0x4470ed, 0x0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/tendermint/go-common/net.go:11 +0xc5
github.com/tendermint/tendermint/vendor/github.com/tendermint/abci/client.(*socketClient).OnStart(0xc4200aefa0, 0x60, 0xc42005e4e0)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/tendermint/abci/client/socket_client.go:68 +0x3e
github.com/tendermint/tendermint/vendor/github.com/tendermint/go-common.(*BaseService).Start(0xc4200aefa0, 0x0, 0xc42005e4e0, 0xc420016400)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/tendermint/go-common/service.go:96 +0x2c2
github.com/tendermint/tendermint/vendor/github.com/tendermint/abci/client.NewSocketClient(0x87bdd3, 0xf, 0x4af201, 0xa6b4a0, 0xc42000e018, 0xc4200b7ea8)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/tendermint/abci/client/socket_client.go:57 +0x23c
github.com/tendermint/tendermint/vendor/github.com/tendermint/abci/client.NewClient(0x87bdd3, 0xf, 0x876c8d, 0x6, 0x1, 0x0, 0xc420011630, 0xc420011600, 0xc420011600)
        /home/vagrant/go/src/github.com/tendermint/tendermint/vendor/github.com/tendermint/abci/client/client.go:49 +0x17a
main.main()
        /home/vagrant/go/src/github.com/tendermint/tendermint/transacter.go:21 +0x1bf
exit status 2
```

`client, err := abciclient.NewClient("tcp://127.0.0.1:46657", "socket", true)` works.

Return an error?

### PR DESCRIPTION
**Pull request by ericdmann**
Closes #72 

Since the included lib (go-common) is deprecated, and updated tmlibs contains the proper check, handle here until new tmlibs can be used exclusively.

https://github.com/tendermint/go-common/blob/47e06734f6ee488cc2e61550a38642025e1d4227/net.go#L11

### CODE REVIEW

File: client/socket_client.go [hunk 1]
```diff
@@ -61,6 +62,10 @@ func (cli *socketClient) OnStart() error {
 		return err
 	}
 
+	if len(strings.SplitN(cli.addr, "://", 2)) < 2 {
+		return fmt.Error("Missing protocol provided (ex. tcp://)", cli.addr)
```
Comment (melekes) [inline client/socket_client.go]: Maybe `Missing protocol (ex. tcp://) in %v`?

**Comment by melekes** [comment]
@ericdmann thank you! Could you please also write a small test for this?

File: client/socket_client.go (hunk 1 above)
Comment (ericdmann) [reply client/socket_client.go]: Love it, will incorporate

**Comment by ericdmann** [comment]
Test added - excuse the extra commit, vendor copy stole the change for a moment :)

**Comment by ebuchman** [comment]
Thanks for this, but not sure I understand. `addr` is only used in `cmn.Connect`, which defaults to `tcp://` if not provided. Why do we need this?

**Comment by ebuchman** [comment]
That said, you've inspired me to do https://github.com/tendermint/abci/commit/6c41ec65bcd86489041fb3bf3a7c56cbe9e70dcf

**Comment by ericdmann** [comment]
The version of Connect() used here (if I'm branched correctly) is from the deprecated go-common (https://github.com/tendermint/go-common/blob/47e06734f6ee488cc2e61550a38642025e1d4227/net.go#L11) and does not check size or default to tcp://, which seems to be at the source of #72. 

This change will protect against the deprecated version until the new tmlibs `Connect` can be used here.

**Comment by melekes** [comment]
Shouldn't we rewrite abci to use tmlibs version then?

**Comment by ebuchman** [comment]
> Shouldn't we rewrite abci to use tmlibs version then?

As far as I can tell it does?

**Comment by ericdmann** [comment]
@ebuchman you're correct, I was working off the trace in the #72 description -- but yes, `tmlibs` solves.

**Pull request closed without merging**